src/app/shell/personal-cabinet/shared-cabinet/messages/messages.component.ts

```diff
@@ -0,0 +1,224 @@
+import { Select, Store } from '@ngxs/store';
+import {
+  debounceTime,
+  distinctUntilChanged,
+  filter,
+  Observable,
+  takeUntil
+} from 'rxjs';
+import { Provider } from '../../../../shared/models/provider.model';
+import { TruncatedItem } from '../../../../shared/models/truncated.model';
+import { ProviderState } from '../../../../shared/store/provider.state';
+import { EntityType, Role } from '../../../../shared/enum/role';
+import { RegistrationState } from '../../../../shared/store/registration.state';
+import {
+  BlockParent,
+  GetWorkshopListByProviderId,
+  UnBlockParent
+} from '../../../../shared/store/provider.actions';
+import { WorkshopDeclination } from '../../../../shared/enum/enumUA/declinations/declination';
+import { ChatRoom } from '../../../../shared/models/chat.model';
+import { Workshop } from '../../../../shared/models/workshop.model';
+import { ConfirmationModalWindowComponent } from '../../../../shared/components/confirmation-modal-window/confirmation-modal-window.component';
+import { Constants } from '../../../../shared/constants/constants';
+import { ModalConfirmationType } from '../../../../shared/enum/modal-confirmation';
+import { BlockedParent } from '../../../../shared/models/block.model';
+import { ReasonModalWindowComponent } from '../applications/reason-modal-window/reason-modal-window.component';
+import { CabinetDataComponent } from '../cabinet-data.component';
+import { MatDialog } from '@angular/material/dialog';
+import { NavBarName } from '../../../../shared/enum/navigation-bar';
+import { PushNavPath } from '../../../..//shared/store/navigation.actions';
+import { GetUserChatRooms } from '../../../..//shared/store/chat.actions';
+import { FormControl } from '@angular/forms';
+import { ChatState } from '../../../../shared/store/chat.state';
+import { SearchResponse } from '../../../../shared/models/search.model';
+import { NoResultsTitle } from '../../../../shared/enum/no-results';
+
+@Component({
+  selector: 'app-messages',
+  templateUrl: './messages.component.html',
+  styleUrls: ['./messages.component.scss']
+})
+export class MessagesComponent extends CabinetDataComponent implements OnInit {
+  readonly Role = Role;
+  readonly WorkshopDeclination = WorkshopDeclination;
+  readonly noMessagesTitle = NoResultsTitle.noMessages;
+
+  userRole: Role;
+  providerId: string;
+  filterFormControl: FormControl = new FormControl('');
+  chatRooms: SearchResponse<ChatRoom[]> = { entities: [], totalAmount: 0 };
+
+  @Select(ProviderState.truncated)
+  workshops$: Observable<TruncatedItem[]>;
+  @Select(RegistrationState.provider)
+  provider$: Observable<Provider>;
+  @Select(ChatState.chatRooms)
+  chatRooms$: Observable<SearchResponse<ChatRoom[]>>;
+
+  constructor(protected store: Store, protected matDialog: MatDialog) {
+    super(store, matDialog);
+  }
+
+  protected init(): void {
+    this.userRole = this.store.selectSnapshot<Role>(RegistrationState.role);
```

Comment:
I think this moment is already handled by CabinetDataComponent 
![image](https://user-images.githubusercontent.com/27493959/197971961-ae10b423-312d-4b0a-934e-921f8538d6af.png)